cumulusci/tasks/apex/custom_settings.py

```diff
@@ -0,0 +1,94 @@
+""" a task for waiting on a specific custom settings value """
+
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+from cumulusci.core.utils import process_bool_arg
+
+
+class CustomSettingValueWait(BaseSalesforceApiTask):
+    """ CustomSettingValueWait polls an org until the specific value exists in a custom settings field """
+
+    name = "CustomSettingValueWait"
+
+    task_options = {
+        "object": {
+            "description": "Name of the SObject to query. Can include the %%%NAMESPACE%%% token. ",
+            "required": True,
+        },
+        "field": {
+            "description": "Name of the field on the Sobject to query. Can include the %%%NAMESPACE%%% token. ",
+            "required": True,
+        },
+        "value": {
+            "description": "Value of the field to wait for (String or Boolean). ",
+            "required": True,
+        },
+        "managed": {
+            "description": (
+                "If True, will inject the project's namespace prefix to replace %%%NAMESPACE%%% tokens in the object or field "
+                "Defaults to False or no namespace."
+            ),
+            "required": False,
+        },
+        "poll_interval": {
+            "description": (
+                "Seconds to wait before polling for batch job completion. "
+                "Defaults to 10 seconds."
+            ),
+        },
+    }
+
+    def _run_task(self):
+        self.poll_interval_s = int(self.options.get("poll_interval", 10))
+
+        self.object_name = self.options["object"]
+        self.field_name = self.options["field"]
+        self.check_value = self.options["value"]
+
+        # Process namespace tokens
+        managed = self.options.get("managed") or False
+        namespace = self.project_config.project__package__namespace
+        namespace_prefix = ""
+        if managed:
+            namespace_prefix = namespace + "__"
+
+        self.object_name = self.object_name.replace("%%%NAMESPACE%%%", namespace_prefix)
+        self.field_name = self.field_name.replace("%%%NAMESPACE%%%", namespace_prefix)
+
+        self._poll()  # will block until poll_complete
+
+        self.logger.info("Value Matched.")
+
+        return True
+
+    def _poll_action(self):
+        query_results = self.sf.query(self._object_query)
+
+        self.record = query_results["records"][0]
```

Comment:
Hmm, so we're just assuming that the first record returned is the one we care about? That doesn't sound like a safe assumption when there is more than one instance of the custom settings object.

Reply:
All of the Hierarchical custom settings objects in NPSP will always have only a single row unless someone manually creates their own individual record. That's extremely rare. The best I can think of to get the org default record is to find the record where the SetupOwnerId starts with "00D". That would have to be done in a separate loop of the returned records. It's feasible to do that if you'd like. Doing this would also limit this to only work on Hierarchical Custom Settings records - preventing the User from trying to use it against any other SObject or even a List-type Custom Settings object.

Reply:
I added logic to verify that the object is a Hierarchical Custom Settings object and to get only the Org Default record if there is more than one record in the object